src/main/java/net/mcreator/blockly/java/blocks/ProcedureRetvalBlock.java

```diff
@@ -24,33 +24,27 @@
 import net.mcreator.element.parts.Procedure;
 import net.mcreator.generator.template.TemplateGeneratorException;
 import net.mcreator.util.XMLUtil;
+import net.mcreator.workspace.elements.VariableElement;
+import net.mcreator.workspace.elements.VariableElementType;
+import org.apache.commons.lang3.StringUtils;
 import org.w3c.dom.Element;
 
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
+import java.util.Set;
 
 public class ProcedureRetvalBlock implements IBlockGenerator {
+	private final Set<String> names = new HashSet<>();
 
-	@Override public void generateBlock(BlocklyToCode master, Element block) throws TemplateGeneratorException {
-		String type;
-
-		String blocktype = block.getAttribute("type");
-		switch (blocktype) {
-		case "procedure_retval_number":
-			type = "NUMBER";
-			break;
-		case "procedure_retval_string":
-			type = "STRING";
-			break;
-		case "procedure_retval_logic":
-			type = "LOGIC";
-			break;
-		case "procedure_retval_itemstack":
-			type = "ITEMSTACK";
-			break;
-		default:
-			return;
+	public ProcedureRetvalBlock() {
+		for (VariableElementType var : VariableElement.getVariables()) {
+			names.add("procedure_retval_" + var.getBlockName());

```

Comment:
Make array at this point and store array in the object field. Same for other IBlockGenerators